state/trackableDataTrie.go

```diff
@@ -193,34 +204,46 @@ func (tdaw *trackableDataTrie) updateTrieWithAutoBalancing() (map[string][]byte,
 	return oldValues, nil
 }
 
-func (tdaw *trackableDataTrie) getOldKeyAndValWithCleanup(key string) ([]byte, []byte, error) {
+func (tdaw *trackableDataTrie) getOldKeyAndValWithCleanup(key string) (common.TrieData, error) {
 	hashedKey := tdaw.hasher.Compute(key)
 
 	oldVal, _, err := tdaw.tr.Get(hashedKey)
 	if err == nil && len(oldVal) != 0 {
-		return hashedKey, oldVal, nil
+		return common.TrieData{
+			Key:     hashedKey,
+			Value:   oldVal,
+			Version: common.AutoBalanceEnabled,
+		}, nil
 	}
 
 	oldVal, _, err = tdaw.tr.Get([]byte(key))
 	if err != nil {
-		return nil, nil, err
+		return common.TrieData{}, err
 	}
 
 	if len(oldVal) == 0 {
-		return hashedKey, oldVal, nil
+		return common.TrieData{
+			Key:     hashedKey,
+			Value:   nil,
+			Version: common.NotSpecified,
+		}, nil
 	}
 
 	err = tdaw.tr.Delete([]byte(key))
 	if err != nil {
-		return nil, nil, err
+		return common.TrieData{}, err
 	}
 
-	return []byte(key), oldVal, nil
+	return common.TrieData{
+		Key:     []byte(key),
+		Value:   oldVal,
+		Version: common.NotSpecified,
+	}, nil
 }
 
-func (tdaw *trackableDataTrie) updateValInTrie(key []byte, val []byte) error {
+func (tdaw *trackableDataTrie) updateValInTrie(key []byte, val []byte, selfDataTrie dataTrie) error {
```

Comment:
i'm thinking if we should consider renaming this `updateValInTrieWithAutobalancing` or something,  to differentiate from v1